### DISCUSSION TITLE
Plugin that `dependsOn` both resolved and unresolved plugins is not loaded

### DISCUSSION THREAD

**Discussion by hitochan777** [nuxt/nuxt#25301] [Questions]
I am new to Nuxt3 (though I have been using Nuxt2) and I am not sure if this is a bug or expected behavior so I am posting as discussion.

### Environment
Node Version: v18.18.0
Nuxt Version: 3.9.3

### Reproduction
https://stackblitz.com/edit/github-pwakuy?file=plugins%2Fb.ts

### Describe the behavior

I have defined three plugins `a.ts`, `b.ts`, `c.ts`.
`b` depends on `a` and `c` so I used `dependsOn` options as follows.
Other plugins are almost the same as plugin `b` except they do not define any dependencies.

```ts
export default defineNuxtPlugin({
  name: 'b',
  dependsOn: ['a', 'c'],
  setup() {
    console.log('setup plugin b');
  },
});
```

I expect plugins to be loaded in the order of `a`, `c`, `b`.
To check if it is loaded in this order I put console.log in setup function.
But when I ran the dev server and server receives a request, it just shows the following in the console log and it seems plugin `b` is not loaded.
If I remove the dependency on `a`, plugin `b` is loaded.

```
setup plugin a
setup plugin c
```

I have read the [Nuxt code on dependsOn](https://github.com/nuxt/nuxt/blob/6736decba0dc8bbc7276848a8a26973440ebdee4/packages/nuxt/src/app/nuxt.ts#L375), and it seems after an plugin is applied it deletes itself from `dependsOn` of unresolved plugins, and only if `dependsOn` becomes empty it applies the unresolved plugins.
But if `dependsOn` contains plugins that are executed earlier than the unresolved plugin, `dependsOn` will never be empty and its plugin will never be applied.
Using the example above, when plugin c is applied it deletes itself from `dependsOn` of plugin b and it becomes `["a"]`, not empty. Thus, plugin b will never be applied.

I am sorry if my understanding is incorrect. But is this expected behavior or a bug?

**Comment by TheAlexLichter** [comment]
Seems like an actual issue. Raised in https://github.com/nuxt/nuxt/issues/25305
